From 705101de9261cf7835e359c3925ff1cee8b8f3e7 Mon Sep 17 00:00:00 2001
From: Mihail Konev <k.mvc@ya.ru>
Date: Sun, 5 Mar 2017 22:58:09 +0500
Subject: [PATCH 2/2] fix current dir inheritance for alt-f2 on msys2

---
 src/child.c   |  3 +++
 src/winmain.c | 14 ++++++++++++++
 2 files changed, 17 insertions(+)

diff --git a/src/child.c b/src/child.c
index 247f4ae..2b92ca9 100644
--- a/src/child.c
+++ b/src/child.c
@@ -1266,6 +1266,9 @@ extern int horsqueeze(void);  // should become horsqueeze_cols in win.h
     if (icon_is_from_shortcut)
       setenv("MINTTY_ICON", cs__wcstoutf(cfg.icon), true);
 
+    // tell alt-f2 mintty instance that it is such one
+    setenv("MINTTY_IS_CHILD", "1", 1);
+
     //setenv("MINTTY_CHILD", "1", true);
 
 #if CYGWIN_VERSION_DLL_MAJOR >= 1005
diff --git a/src/winmain.c b/src/winmain.c
index ba7925f..4122c68 100644
--- a/src/winmain.c
+++ b/src/winmain.c
@@ -5504,6 +5504,12 @@ opts[] = {
   {0, 0, 0, 0}
 };
 
+static int getenvi(const char *varname) {
+  char *s = getenv(varname);
+  if (s == NULL) { return 0; }
+  return atoi(s);
+}
+
 int
 main(int argc, char *argv[])
 {
@@ -6865,6 +6871,14 @@ static int dynfonts = 0;
     launcher_setup_argv();
   }
 
+  // Ask /etc/post-install/05-home-dir.post not to "cd ~/" in an Alt-F2 window
+  {
+    if (getenvi("MINTTY_IS_CHILD")) {
+      putenv("CHERE_INVOKING=1");
+    }
+    unsetenv("MINTTY_IS_CHILD");
+  }
+
   // Create child process.
   /* We could move this below SetFocus() or win_init_drop_target() 
      in order to further reduce the delay until window display (#1113) 
-- 
2.35.3

