From 2e23662f7eb4f6fcdec4492634218ebed5659676 Mon Sep 17 00:00:00 2001
From: Takashi Yano <takashi.yano@nifty.ne.jp>
Date: Sat, 5 Feb 2022 16:52:33 +0900
Subject: [PATCH 004/N] Cygwin: console: Remove unnecessary (redundant) code.

Signed-off-by: Johannes Schindelin <johannes.schindelin@gmx.de>
---
 winsup/cygwin/fhandler_console.cc | 2 --
 1 file changed, 2 deletions(-)

diff --git a/winsup/cygwin/fhandler_console.cc b/winsup/cygwin/fhandler_console.cc
index 65325ad..23421db 100644
--- a/winsup/cygwin/fhandler_console.cc
+++ b/winsup/cygwin/fhandler_console.cc
@@ -2177,7 +2177,6 @@ fhandler_console::char_command (char c)
 	      if (y == con.b.srWindow.Bottom)
 		{
 		  WriteConsoleW (get_output_handle (), L"\033[2K", 4, 0, 0);
-		  wpbuf.empty ();
 		  break;
 		}
 	      if (y == con.b.srWindow.Top
@@ -2222,7 +2221,6 @@ fhandler_console::char_command (char c)
 	      if (y == con.b.srWindow.Bottom)
 		{
 		  WriteConsoleW (get_output_handle (), L"\033[2K", 4, 0, 0);
-		  wpbuf.empty ();
 		  break;
 		}
 	      __small_swprintf (bufw, L"\033[%d;%dr",
-- 
2.9.0

