From 6db22b6ff09464c5b5bb1ef86bee4c19244c6a74 Mon Sep 17 00:00:00 2001
From: Takashi Yano <takashi.yano@nifty.ne.jp>
Date: Fri, 18 Mar 2022 23:28:36 +0900
Subject: [PATCH 053/N] Cygwin: console: Add attach_mutex guard that was not
 added.

Signed-off-by: Johannes Schindelin <johannes.schindelin@gmx.de>
---
 winsup/cygwin/fhandler_console.cc | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/winsup/cygwin/fhandler_console.cc b/winsup/cygwin/fhandler_console.cc
index b52ecfc..16850ae 100644
--- a/winsup/cygwin/fhandler_console.cc
+++ b/winsup/cygwin/fhandler_console.cc
@@ -361,7 +361,9 @@ remove_record:
 		  {
 		    if (total_read + j - i >= n)
 		      { /* Something is wrong. Giving up. */
+			acquire_attach_mutex (mutex_timeout);
 			WriteConsoleInputW (p->input_handle, tmp, n, &n);
+			release_attach_mutex ();
 			goto skip_writeback;
 		      }
 		    input_rec[total_read + j - i] = tmp[j];
-- 
2.9.0

